### PR TITLE
Feature add managers in cell

### PR DESCRIPTION
**Pull request by RignonNoel**
| Q                                                      | R
| ------------------------------------------ | -------------------------------------------
| Type of contribution ?                      | Features
| Tickets (_issues_) concerned               | #155 

---

##### What have you changed ?
Add a way to link users with a cell as a `manager`.

##### How did you change it ?
1. Add a M2M field `managers` in `Cell` model
2. Refactor `create()`, `update()` and `to_representation()` of the cell's serializer
3. Create `UserPublicSerializer` to represent manager without confidential information
4. Update tests

##### Verification :

**PR standard**
 - [x] Branches naming convention: 
     - `prefix-snake_case_description`
     - ex: `enhancement-create_new_cell`
     - ex: `fix-pep8_standard_on_cell_model`
 - [x] Fill in this automatically generated PR template
 - [x] Do not include issue numbers in the PR title
 - [x] Include screenshots and animated GIFs in your pull request whenever possible

**Code standard**
 - [x] This Pull-Request fully meets the requirements defined in the issue
 - [x] Add or modify the attached tests
 - [x] Follow the Python Styleguide
 - [x] Document new code based on the Documentation Styleguide
 - [ ] Use I18N functions when you add some text
  - I didn't use I18N function because the actual serializer doesn't use it. We need to discuss and create an issue to refactor I18N on all this serializer.

### CODE REVIEW

**Comment by yanicolivier** [comment]
I suggest adding to the admin this small portion of code to have the nice horizontal_filter:
 
`class CellAdmin(admin.ModelAdmin):
    list_display = [
        'name',
    ]
    
    filter_horizontal = ('managers',)

admin.site.register(models.Cell, CellAdmin)
`

File: source/apiVolontaria/volunteer/serializers.py [hunk 1]
```diff
@@ -98,65 +107,114 @@ def create(self, validated_data):
...
+    def update(self, instance, validated_data):
+        if 'name' in validated_data.keys():
+            instance.name = validated_data['name']
+        if 'managers' in validated_data.keys():
+            # Remove all managers of the cell
+            instance.managers.clear()
+
+            for manager in validated_data['managers']:
+                try:
+                    user = User.objects.get(id=manager)
```
Comment (yanicolivier) [inline source/apiVolontaria/volunteer/serializers.py]: the output of manager is for example : 
`{'id': 8, 'username': 'root', 'first_name': '', 'last_name': '', 'email': 'a@b.com'}`

so this line should be : 
`user = User.objects.get(id=manager['id'])`

Comment (RignonNoel) [reply source/apiVolontaria/volunteer/serializers.py]: I agree with you the output of a manager is :
```
{'id': 8, 'username': 'root', 'first_name': '', 'last_name': '', 'email': 'a@b.com'}
```

But his input is just one or multiple IDs:
```
managers: [
    31,
]
```

**Comment by RignonNoel** [comment]
> I suppose we will add a validation after to know if the user is a cell manager to be able to add to the manager. We don't want to permit any user. @yanicolivier 

A user become a cell manager when he become a manager of a cell (by definition). We want to permit any user to become a cell manager depending on what the administrator want to do, only an administrator can update a cell and change managers.

File: source/apiVolontaria/volunteer/serializers.py (hunk 1 above)
Comment (yanicolivier) [reply source/apiVolontaria/volunteer/serializers.py]: Okay, make sense.

**Pull request merged**